### PR TITLE
chore(docs): include the environment variables section in the utilities documentation

### LINKED ISSUE #2923: Docs: Include environment variables in Tracer
### What were you searching in the docs?

Environment variables I can use with Tracer

### Is this related to an existing documentation section?

https://docs.powertools.aws.dev/lambda/python/latest/core/tracer/

### How can we improve?

Introduce a new subsection `Environment variables` in the Tracer docs, so customers don't have to have two tabs opened, or miss this is possible.

https://docs.powertools.aws.dev/lambda/python/latest/#environment-variables

### Got a suggestion in mind?

Proposal graciously done by [@barreeeiroo](https://github.com/barreeeiroo) already

https://github.com/aws-powertools/powertools-lambda-python/pull/2922

### Acknowledgment

- [X] I understand the final update might be different from my proposed suggestion, or refused.

### PR DESCRIPTION
**Pull request by barreeeiroo**
**Issue number:** https://github.com/aws-powertools/powertools-lambda-python/issues/2923
**Supersedes:** https://github.com/aws-powertools/powertools-lambda-python/pull/2922

## Summary

Clarifies some documentation pages by including their related environment variables. They were only included on the main page, though it may be useful to also mention them on their corresponding page.

### Changes

For each page, a new _Environment variables_ section was created under _Getting started_ right before _Advanced_. This section contains a table listing all environment variables affecting that page. It also mentions the preference order (parameter overrides environment variable).

### User experience

When navigating through documentation, developers will now find the environment variables for each core/utility on its page. It is useful as it allows us to quickly understand the behaviour of these environment variables by <kbd>Ctrl</kbd>+<kbd>F</kbd>'ing them.

This comes from personal experience, as I found out about Tracer's environment variables by checking the parameters of Tracer decorators' source code. But when going to Tracer's documentation page, I couldn't find these environment variables, until I realized they were only in the main page.

## Checklist

If your change doesn't seem to apply, please leave them unchecked.

* [x] [Meet tenets criteria](https://docs.powertools.aws.dev/lambda/python/#tenets)
* [x] I have performed a self-review of this change
* [x] Changes have been tested
* [x] Changes are documented
* [x] PR title follows [conventional commit semantics](https://github.com/aws-powertools/powertools-lambda-python/blob/develop/.github/semantic.yml)

<details>
<summary>Is this a breaking change?</summary>

No.

Checklist:

* [ ] Migration process documented
* [ ] Implement warnings (if it can live side by side)

</details>

## Acknowledgment

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

**Disclaimer**: We value your time and bandwidth. As such, any pull requests created on non-triaged issues might not be successful.

### CODE REVIEW

**Comment by leandrodamascena** [comment]
Thanks for sending this PR @barreeeiroo!!

Hei @heitorlessa! Since I don't have the context here, I'd like your review or we can sync tomorrow and I can review.

File: docs/utilities/middleware_factory.md [hunk 1]
```diff
@@ -19,6 +19,16 @@ Middleware factory provides a decorator factory to create your own middleware to
 
 You might need a custom middleware to abstract non-functional code. These are often custom authorization or any reusable logic you might need to run before/after a Lambda function invocation.
 
+### Environment variables
```
Comment (leandrodamascena) [inline docs/utilities/middleware_factory.md]: What do you think of the "Environment Variables" section being the last of the Getting Started section? So we keep a standard.

File: docs/core/logger.md [hunk 2]
```diff
@@ -341,6 +341,18 @@ You can easily change the date format using one of the following parameters:
     --8<-- "examples/logger/src/date_formatting_output.json"
     ```
 
+### Environment variables
```
Comment (leandrodamascena) [inline docs/core/logger.md:346]: In logger.md we already have a section that explains some ENV variables (https://docs.powertools.aws.dev/lambda/python/2.22.0/core/logger/#getting-started). How about adding a disclaimer like: "check for additional environment variables here" and adding a link to this new section?

**Review by leandrodamascena — CHANGES_REQUESTED** [review]
Hello @barreeeiroo! I made a couple of small comments and after that I think we're ready to merge!

Thank you very much.

File: docs/utilities/middleware_factory.md (hunk 1 above)
Comment (barreeeiroo) [reply docs/utilities/middleware_factory.md]: Yeah, my bad, I was actually placing all of them at the end of _Getting Started_ right before _Advanced_, but I probably missed that one.

File: docs/core/logger.md (hunk 2 above)
Comment (barreeeiroo) [reply docs/core/logger.md:346]: Yep, that makes sense. Will add it 👍
I've also seen that Metrics also mentions a few environment variables for the quick start, though it is only 1 environment variable so there are no other variables in the corresponding section.

File: docs/utilities/middleware_factory.md (hunk 1 above)
Comment (leandrodamascena) [reply docs/utilities/middleware_factory.md]: Thanks for addressing this feedback @barreeeiroo! :rocket:

**Review by leandrodamascena — APPROVED** [review]
Fantastic work @barreeeiroo! This will make it easier for our customers to find environment variables that can be used in each utility.

image

**Pull request merged**